devenv/lib/fs.py

```diff
@@ -1,15 +1,25 @@
 from __future__ import annotations
 
 import os
+import subprocess
 
 from devenv.constants import home
 from devenv.constants import shell
 from devenv.lib import proc
 
 
+def zdotdir() -> str:
+    return subprocess.run(
+        [shell, "-c", "echo $ZDOTDIR"], text=True, capture_output=True
+    ).stdout.strip()
+
+
 def shellrc() -> str:
     if shell == "zsh":
-        return f"{home}/.zshrc"
+        dotdir = zdotdir()
```

Comment:
I'd check for `~/.zshrc` to exist first so we don't have an extra subprocess for the common case.

Could you also add your PR description as a comment? It's not obvious.